pkg/doc/did/doc.go

```diff
@@ -26,18 +27,147 @@ const (
+  "required": [
+    "@context",
+    "id"
+  ],
+  "properties": {
+    "@context": {
+      "type": "array",
+      "items": [
+        {
+          "type": "string",
+          "pattern": "^https://w3id.org/did/v1$"
+        }
+      ],
+      "additionalItems": {
+        "type": "string",
+        "format": "uri"
+      }
+    },
+    "id": {
+      "type": "string"
+    },
+    "publicKey": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/publicKey"
+      }
+    },
+    "authentication": {
+      "type": "array",
+      "items": {
+        "oneOf": [
+          {
+            "$ref": "#/definitions/publicKey"
+          },
+          {
+            "type": "string"
+          }
+        ]
+      }
+    },
+    "service": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/service"
+      }
+    },
+    "created": {
+      "type": "string",
+      "pattern": "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\dZ"
+    },
+    "updated": {
+      "type": "string",
+      "pattern": "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\dZ"
+    },
+    "proof": {
+      "type": "object",
+      "required": [ "type", "creator", "created", "signatureValue"],
+      "properties": {
+        "type": {
+          "type": "string",
+          "format": "uri-reference"
+        },
+        "creator": {
+          "type": "string",
+          "format": "uri-reference"
+        },
+        "created": {
+          "type": "string",
+          "pattern": "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\dZ"
+        },
+        "signatureValue": {
+          "type": "string"
+        },
+        "domain": {
+          "type": "string"
+        },
+        "nonce": {
+          "type": "string"
+        }
+  }
+    }
+  },
+  "definitions": {
+    "publicKey": {
+      "required": [
+        "id",
+        "type",
+        "controller"
+      ],
+      "type": "object",
+      "minProperties": 4,
+      "maxProperties": 4,
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "controller": {
+          "type": "string"
+        }
+      }
+    },
+    "service": {
+      "required": [
+        "id",
+        "type",
+        "serviceEndpoint"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "serviceEndpoint": {
+          "type": "string",
+          "format": "uri"
+        }
+      }
+    }
+  }
+}`
 )
 
+var schemaLoader = gojsonschema.NewStringLoader(schema)
+
 // Doc DID Document definition
 type Doc struct {
 	Context        []string
 	ID             string
 	PublicKey      []PublicKey
 	Service        []Service
 	Authentication []VerificationMethod
-	Created        time.Time
-	Updated        time.Time
-	Proof          Proof
+	Created        *time.Time
```

Comment:
Why does this need to be a pointer?

Reply:
@troyronda it's an optional field: https://w3c-ccg.github.io/did-spec/#created-optional